algorithms/soluciones/valid_parentheses.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+"""Valid_Parentheses.ipynb
+
+Automatically generated by Colab.
+
+Original file is located at
+    https://colab.research.google.com/drive/15tqvKTOrmKdX3n536yLORVmC_gNL8vSG
+"""
+
+def valid(s: str):
+    if len(s) >= 1 and len(s) <= 104:
+        per = [")", "(", "]", "[", "}", "{"]
+        if all(car in per for car in s):
```

Comment:
you are using two loops to read all the strings, try to read the string just once. 😸